packages/playground/blueprints/src/lib/compile.ts

```diff
@@ -85,6 +85,9 @@ export function compileBlueprint(
 		onStepCompleted = () => {},
 	}: CompileBlueprintOptions = {}
 ): CompiledBlueprint {
+	// Deep clone the blueprint to avoid mutating the input
+	blueprint = JSON.parse(JSON.stringify(blueprint));
```

Comment:
Real question: Is there a reason we shouldn't use `structuredClone()` here?

Reply:
The only reason I didn't use it was I forgot it exists, thanks :D